certification/pyxis/pyxis_suite_test.go

```diff
@@ -77,13 +81,26 @@ func (p *pyxisImageHandler) ServeHTTP(response http.ResponseWriter, request *htt
 	if request.Body != nil {
 		defer request.Body.Close()
 	}
+	responseString := `{"_id":"blah","certified":false,"deleted":false,"image_id":"123456789abc"}`
+	log.Tracef("Method: %s", request.Method)
 	switch {
-	case strings.Contains(request.URL.Path, "my-image-409-project-id") && request.Method == http.MethodPost:
-		response.WriteHeader(409)
+	case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-image-409-api-token":
+		response.WriteHeader(http.StatusConflict)
+	case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-401-image-api-token":
+		response.WriteHeader(http.StatusConflict)
+	case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-image-api-token":
+		response.WriteHeader(http.StatusConflict)
+	case request.Method == http.MethodGet && request.Header["X-Api-Key"][0] == "my-bad-401-image-api-token":
+		log.Trace("get with 401")
+		response.WriteHeader(http.StatusUnauthorized)
 	case request.Header["X-Api-Key"][0] == "my-bad-image-api-token":
-		response.WriteHeader(401)
+		response.WriteHeader(http.StatusUnauthorized)
+	case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-500-image-api-token":
+		response.WriteHeader(http.StatusInternalServerError)
+	case request.Method == http.MethodPost:
+		mustWrite(response, responseString)
```

Comment:
This looks like it intends to test a variety of scenarios. I'd benefit from a few words in a comment going over the matrix.

Reply:
How do you feel about nested switch? I think it might help the clarity here. I'm thinking switching on request.Method as the top-level.

Reply:
I'm on the fence. But I'm also somewhat on the fence with the general need for a switch here.

I kind of get the need to do multiple scenarios. I think what I'm wondering is, would there be a way to restructure the test that would require less switching?

Reply:
I take it back. I think I want to stick with a flat switch. Then I don't end up making duplicates for ones that are the same for Get and Post.

Reply:
FWIW, when I was adding tests to this, I thought about a nested switch as well when I thought I might need to marshal the request payload for an edge case, but when I coded that I hated how it read. I ended up just going with an `if` for the one case, and default return for everything else. Hopefully we never have a case were need more then two options within a given switch (ie GET vs POST).